// @fitbit/sdk/lib/capabilities.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SupportedDeviceCapabilities = exports.DeviceCapability = void 0;
const tslib_1 = require("tslib");
const buildTargets_1 = tslib_1.__importDefault(require("./buildTargets"));
var DeviceCapability;
(function (DeviceCapability) {
    DeviceCapability["SCREEN_SIZE"] = "screenSize";
})(DeviceCapability = exports.DeviceCapability || (exports.DeviceCapability = {}));
var SupportedDeviceCapabilities;
(function (SupportedDeviceCapabilities) {
    function create(targetDevice) {
        var _a;
        const specs = (_a = buildTargets_1.default[targetDevice]) === null || _a === void 0 ? void 0 : _a.specs;
        return specs
            ? {
                [DeviceCapability.SCREEN_SIZE]: {
                    w: specs.screenSize.width,
                    h: specs.screenSize.height,
                },
            }
            : undefined;
    }
    SupportedDeviceCapabilities.create = create;
})(SupportedDeviceCapabilities = exports.SupportedDeviceCapabilities || (exports.SupportedDeviceCapabilities = {}));
